### PR TITLE
Implement upgrade command and update CLI for improved package management

### PR DESCRIPTION
**Pull request by itlackey**
This pull request removes the `reinstall` command and replaces it with an improved `update --force` option, and introduces a new `upgrade` command for self-updating the `akm` binary. Documentation, API, and CLI code are updated to reflect these changes, and new logic is added for robust binary upgrades, including checksum verification and platform-specific handling.

**Registry update and reinstall changes:**

* The `reinstall` command and related API/types are removed; the `update` command now supports a `--force` flag to force a fresh download and extraction of kits, replacing all previous reinstall functionality. [[1]](diffhunk://#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abL16-R20) [[2]](diffhunk://#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abL91-R94) [[3]](diffhunk://#diff-8d492467f5bd676913420c97c86f34466a6eb9c43838f8c558547d1c7c94a534L165-R167) [[4]](diffhunk://#diff-5a8820d0ecca92c000be8d05c4cfda1d6658116a61255a9715165d58b65856f6L313-R313) [[5]](diffhunk://#diff-fa8d4e24d8399e8350f1c8bad05df53e8032ea995835bf911507015e2db61cddR249-R273) [[6]](diffhunk://#diff-ddf675e581ef98e0361929b9fe35d7239e57735833a052217380bd0c819da50fL76-R92) [[7]](diffhunk://#diff-49b4c00dfd11b5046fc995fcdb7ce262e1da6f604532502890dda8246010f208L149-L173)
* Documentation is updated to clarify the new `update --force` workflow and remove references to `reinstall`. [[1]](diffhunk://#diff-27c8c22b45605a0ddf42ce8017b6b2a68cb3b10534c18bf4a781e230d7b92361R152-R178) [[2]](diffhunk://#diff-8d492467f5bd676913420c97c86f34466a6eb9c43838f8c558547d1c7c94a534L266-R264)

**Self-upgrade functionality:**

* Adds a new `upgrade` command to the CLI, with supporting API and types, enabling users to check for updates and upgrade the `akm` binary in-place. [[1]](diffhunk://#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abL16-R20) [[2]](diffhunk://#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abL91-R94) [[3]](diffhunk://#diff-fa8d4e24d8399e8350f1c8bad05df53e8032ea995835bf911507015e2db61cddR249-R273) [[4]](diffhunk://#diff-fa8d4e24d8399e8350f1c8bad05df53e8032ea995835bf911507015e2db61cddL512-R546) [[5]](diffhunk://#diff-1d9cfc4674f6bdabc3276ff78e9cc584a0665dd5390569ce5f7e5d953569a5feR1-R223) [[6]](diffhunk://#diff-49b4c00dfd11b5046fc995fcdb7ce262e1da6f604532502890dda8246010f208R209-R225)
* Implements robust platform-specific binary replacement logic, including checksum verification and error handling for permission issues.

**API and type exports:**

* Removes `agentikitReinstall`, `ReinstallResponse`, and related types from exports; adds `checkForUpdate`, `performUpgrade`, `UpgradeCheckResponse`, and `UpgradeResponse` to API and type exports. [[1]](diffhunk://#diff-dcdc3e0b3362edb8fec2a51d3fa51f8fb8af8f70247e06d9887fa934834c9122L6-R10) [[2]](diffhunk://#diff-dcdc3e0b3362edb8fec2a51d3fa51f8fb8af8f70247e06d9887fa934834c9122L29-R35) [[3]](diffhunk://#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abL164-R171)

**CLI formatting improvements:**

* Updates CLI output formatting to reflect new command responses for `update` and `upgrade`, including per-entry update details and upgrade status messages.

**Internal refactoring:**

* Cleans up references to removed types and functions across the codebase, ensuring consistency and clarity.

### CODE REVIEW

File: tests/self-update.test.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+  })
+
+  test("updateAvailable is true for an old version", async () => {
+    const { checkForUpdate } = await import("../src/self-update")
+    const result = await checkForUpdate("0.0.0")
+
+    expect(result.updateAvailable).toBe(true)
+    expect(result.currentVersion).toBe("0.0.0")
+  })
```
[the remark this answers is not in the document]
Comment (itlackey) [reply tests/self-update.test.ts]: Fixed. All `checkForUpdate` tests now mock `globalThis.fetch` with deterministic payloads instead of making real GitHub API calls. Also added tests for edge cases (missing `tag_name`, non-OK response).

File: tests/e2e.test.ts [hunk 2]
```diff
@@ -807,6 +807,57 @@ describe("Scenario: Registry lifecycle CLI (no network)", () => {
...
+
+  test("cli: akm upgrade detects non-binary install and shows guidance", async () => {
+    // When run via `bun run`, akm is not a compiled binary
+    const result = runCli("upgrade")
+    expect(result.exitCode).toBe(0)
+    const json = parseJson(result.stdout)
+    expect(json.upgraded).toBe(false)
+    // Should be npm or unknown when running via bun
+    expect(["npm", "unknown"]).toContain(json.installMethod)
+  })
```
[the remark this answers is not in the document]
Comment (itlackey) [reply tests/e2e.test.ts]: Fixed. Replaced subprocess-based upgrade tests (which spawned CLI processes hitting GitHub API) with in-process tests using `withMockedFetch`. The remaining subprocess tests (`--help` flag checks, `--force` validation) don't make network calls.
